src/OrbitQt/CaptureOptionsDialog.cpp

```diff
@@ -77,4 +79,36 @@ void CaptureOptionsDialog::ShowSourcePathsMappingEditor() {
   }
 }
 
+void CaptureOptionsDialog::SetCollectMemoryInfo(bool collect_memory_info) {
+  ui_->collectMemoryInfoCheckBox->setChecked(collect_memory_info);
+}
+
+bool CaptureOptionsDialog::GetCollectMemoryInfo() const {
+  return ui_->collectMemoryInfoCheckBox->isChecked();
+}
+
+void CaptureOptionsDialog::SetMemorySamplingPeriodNs(uint64_t memory_sampling_period_ns) {
+  ui_->memorySamplingPeriodNsLineEdit->setText(QString::number(memory_sampling_period_ns));
+}
+
+uint64_t CaptureOptionsDialog::GetMemorySamplingPeriodNs() const {
+  CHECK(!ui_->memorySamplingPeriodNsLineEdit->text().isEmpty());
+  bool valid = false;
+  uint64_t result = ui_->memorySamplingPeriodNsLineEdit->text().toULongLong(&valid);
+  CHECK(valid);
+  return result;
+}
+
+void CaptureOptionsDialog::SetMemoryWarningThresholdKb(uint64_t memory_warning_threshold_kb) {
+  ui_->memoryWarningThresholdKbLineEdit->setText(QString::number(memory_warning_threshold_kb));
+}
+
+uint64_t CaptureOptionsDialog::GetMemoryWarningThresholdKb() const {
+  CHECK(!ui_->memoryWarningThresholdKbLineEdit->text().isEmpty());
+  bool valid = false;
+  uint64_t result = ui_->memoryWarningThresholdKbLineEdit->text().toULongLong(&valid);
+  CHECK(valid);
```

Comment:
same here

Reply:
Thanks for pointing out this. For those line edit components, we only allow the digit inputs by adding `<set>Qt::ImhDigitsOnly</set>`.